cmd/tether/attach.go

```diff
@@ -51,11 +52,18 @@ type attachServerSSH struct {
 	m sync.Mutex
 
 	// conn is held directly as it's how we stop the attach server
-	conn      *net.Conn
+	conn struct {
+		sync.Mutex
```

Comment:
Reading the code, I think i understand the lock contract: Enabled, conn lock, and Disabled, con unlock. I think :-). Can you add a (big honking) comment about the lock order?